applications/fishing-map/src/features/map/Map.tsx

```diff
@@ -115,6 +146,11 @@ const Map = (): React.ReactElement => {
         <IconButton icon="ruler" type="map-tool" tooltip="Ruler (Coming soon)" />
         <IconButton icon="camera" type="map-tool" tooltip="Capture (Coming soon)" />
       </div>
+      {layersWithLegend?.map(
+        (legend) =>
+          document.getElementById(legend.id) &&
+          createPortal(<MapLegend layer={legend} />, document.getElementById(legend.id) as Element)
```

Comment:
This is a very good idea! simplify a lot the needs of sharing the derived styled state! 👏